heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -153,39 +153,41 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       deactivateTopology(stateManager, topology, proposedPackingPlan);
     }
 
-    Set<PackingPlan.ContainerPlan> updatedContainers =
-        new HashSet<>(proposedPackingPlan.getContainers());
-    // request new resources if necessary. Once containers are allocated we should make the changes
-    // to state manager quickly, otherwise the scheduler might penalize for thrashing on start-up
-    if (newContainerCount > 0 && scalableScheduler.isPresent()) {
-      Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
-      Set<PackingPlan.ContainerPlan> containersAdded =
-          scalableScheduler.get().addContainers(containersToAdd);
-      // Update the PackingPlan with new container-ids
-      if (containersAdded != null) {
-        updatedContainers.removeAll(containersToAdd);
-        updatedContainers.addAll(containersAdded);
+    try {
+      Set<PackingPlan.ContainerPlan> updatedContainers =
+          new HashSet<>(proposedPackingPlan.getContainers());
+      // request new resources if necessary. Once containers are allocated we should make the changes
+      // to state manager quickly, otherwise the scheduler might penalize for thrashing on start-up
+      if (newContainerCount > 0 && scalableScheduler.isPresent()) {
+        Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
+        Set<PackingPlan.ContainerPlan> containersAdded =
+            scalableScheduler.get().addContainers(containersToAdd);
+        // Update the PackingPlan with new container-ids
+        if (containersAdded != null) {
+          updatedContainers.removeAll(containersToAdd);
+          updatedContainers.addAll(containersAdded);
+        }
+      }
+  
+      PackingPlan updatedPackingPlan =
+          new PackingPlan(proposedPackingPlan.getId(), updatedContainers);
+      PackingPlanProtoSerializer serializer = new PackingPlanProtoSerializer();
+      PackingPlans.PackingPlan updatedProtoPackingPlan = serializer.toProto(updatedPackingPlan);
+      LOG.fine("The updated Packing Plan: " + updatedProtoPackingPlan);
+  
+      // update packing plan to trigger the scaling event
+      logInfo("Update new PackingPlan: %s",
+          stateManager.updatePackingPlan(updatedProtoPackingPlan, topologyName));
+    } finally {
+      // reactivate topology
+      if (initiallyRunning) {
```

Comment:
Would be much better if we can print meaningful log to notify users when unrecoverable bad things happen, like:
```
Failing to update topology. The topology can be in a strange stage. Please check carefully or redeploy the topology.
```

Reply:
The update procedure may throw several exceptions which are hard to be differentiated in the upper function. I added the suggested print msg in the present exception msg.